[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29494 [quant][graphmode] Returning axis from calculate_qparams
* #29493 [quant][graphmode] Test for per channel graph mode quantization
* **#29416 [quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`**

Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580906](https://our.internmc.facebook.com/intern/diff/D18580906)

jerryzh168 committed:
[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`

Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good!

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on test/test_quantization.py:
```diff
@@ -766,30 +768,38 @@ def test_observer_with_ignored_function(self):
+            qconfig_dict = {
+                '': qconfig
+            }
+            model_script = quantize_script(
```
If I use prepare_script(), will the output be serializable? I am wondering because part of the observer is in python and part is scripted.

jerryzh168 replied:
no, observers with ignored functions are not serializable, do we want to serialize observers?

jerryzh168 replied:
oh, looks like they are still serializable..

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580906](https://our.internmc.facebook.com/intern/diff/D18580906)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Make PerChannelMinMaxObserver scriptable using `torch.jit.ignore`"


Summary:
att

Test Plan:
python test/test_quantization.py

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D18580906](https://our.internmc.facebook.com/intern/diff/D18580906)

[ghstack-poisoned]
